src/DynamoCore/Graph/Nodes/NodeModelExtensions.cs

```diff
@@ -106,6 +107,15 @@ internal static string GetOriginalName(this NodeModel node)
             return nodeType.FullName;
         }
 
+        internal static string GetCustomNodeOriginalName(this NodeModel node)
+        {
+            if (node == null) 
+                return string.Empty;
+
+            var customNodeFunction = node as Function;
```

Comment:
what if this node is not a customNode?

Reply:
if you have thinking of tests to write - this would be a good one since you caught it manually it would be good to load a custom node which has this naming issue and check that your function returns correctly when passed that node.